app/controllers/application_controller.rb

```diff
@@ -60,4 +60,15 @@ def basecamp_auth_check
       @basecamp_agent = current_user.agents.where(type: 'Agents::BasecampAgent').first
     end
   end
+
+  def agent_params
+    return {} unless params[:agent]
+    @agent_params ||= begin
+      options = params[:agent].delete(:options) if params[:agent][:options].present?
```

Comment:
Oh, that makes sense! Should we spec that users cannot set the admin bit in
the devise users controller, or agent user_id in agents controller?
On Sat, Sep 10, 2016 at 11:03 AM Dominik Sander <notifications@github.com>
wrote:

> In app/controllers/application_controller.rb
> <https://github.com/cantino/huginn/pull/1679#discussion_r78278545>:
>
> > @@ -60,4 +60,15 @@ def basecamp_auth_check
> >        @basecamp_agent = current_user.agents.where(type: 'Agents::BasecampAgent').first
> >      end
> >    end
> > +
> > +  def agent_params
> > +    return {} unless params[:agent]
> > +    @agent_params ||= begin
> > +      options = params[:agent].delete(:options) if params[:agent][:options].present?
>
> For calls that do not send the agent options (agent disable for example)
> we can not set it to anything, otherwise the validations will fail. See this
> spec
> <https://github.com/cantino/huginn/blob/master/spec/controllers/agents_controller_spec.rb#L315>
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/cantino/huginn/pull/1679/files/c26adc5970da6ff91d9765afad4209d74a6100fa#r78278545>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAFHe6wjqOWoDqg_kg5v6fRtE6Doa_A-ks5qosbBgaJpZM4J5tpp>
> .
>